app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Model} from '@nozbe/watermelondb';
+import {DeviceEventEmitter} from 'react-native';
+
+import {localMyChannels, markChannelAsUnread, markChannelAsViewed} from '@actions/local/channel';
+import {removePostById} from '@actions/local/post';
+import {fetchMyChannel, markChannelAsRead} from '@actions/remote/channel';
+import {fetchPostAuthors} from '@actions/remote/post';
+import {ActionType, Events, WebsocketEvents} from '@constants';
+import DatabaseManager from '@database/manager';
+import {queryMyChannel} from '@queries/servers/channel';
+import {queryPostById} from '@queries/servers/post';
+import {queryCurrentChannelId, queryCurrentUserId} from '@queries/servers/system';
+import {isFromWebhook, isSystemMessage, shouldIgnorePost} from '@utils/post';
+
+import type {WebSocketMessage} from '@typings/api/websocket';
+
+export async function handleNewPostEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const postFromData = JSON.parse(msg.data.post);
+    const currentUserId = await queryCurrentUserId(database.database);
+    const post: Post = {
+        ...postFromData,
+        ownPost: postFromData.user_id === currentUserId,
+    };
+
+    const existing = await queryPostById(database.database, post.pending_post_id);
+
+    if (existing) {
+        return;
+    }
+
+    database.operator.handlePosts({
```

Comment:
if you want to be able to batch this, we would need to modify the handler, add a `prepareRecordsOnly` as we do with the other ones.. so the how is by adding the functionality :P

I don; think that is really needed here

Reply:
btw `prepareRecordsOnly` has been added in #5893 and then the answer would be yes! let's batch it! in fact, let's aim to batch everything in this action

Reply:
I decided to await for this, since now it also updates the mychannelmember table (to centralize the modification of lastPostAt), and if I remember correctly, two different changes on the same model cannot be in the same batch.

Reply:
The same row on the same batch cannot be done, but you could just update every property in the row that you are updating